### PR TITLE
[SYCLomatic] Optimize the error report of query-api-mapping. 

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Review by zhiweij1 — APPROVED** [review]

File: clang/test/dpct/query_api_mapping/test.cu [hunk 1]
```diff
@@ -17,3 +17,14 @@
...
+// RUN: not dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=aaa 2>&1 | FileCheck %s -check-prefix=NO_MAPPING
+
+// NO_MAPPING: dpct exited with code: -43 (Error: This API mapping query is not available yet. You may get the API mapping by migrating a sample code of this API with the tool.)
+
+// RUN: not dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=hdiv 2>&1 | FileCheck %s -check-prefix=PARSE_FAILED
+
+// PARSE_FAILED: {{(.+)}}temp.cu:7:3: error: use of undeclared identifier 'hdiv'
+// PARSE_FAILED-NEXT:     7 |   hdiv(h1 /*__half*/, h2 /*__half*/);
+// PARSE_FAILED-NEXT:       |   ^
+// PARSE_FAILED-NEXT: dpct exited with code: -17 (Error: Cannot parse input file(s))
```
Comment (zhimingwang36) [inline clang/test/dpct/query_api_mapping/test.cu:30]: Please further update the L27~L30 error processing.  For end user, it doesn't make sense to see those line ,they want to know the hint why mapping not available:  1. API is not available in current CUDA header files, please use -cuda-sdk-header to provide 2.  API declare header file is not available.

Comment (zhimingwang36) [reply clang/test/dpct/query_api_mapping/test.cu:30]: Please update it in separate PR.

**Review by zhimingwang36 — APPROVED** [review]
LGTM

File: clang/test/dpct/query_api_mapping/test.cu (hunk 1 above)
Comment (tangjj11) [reply clang/test/dpct/query_api_mapping/test.cu:30]: Will fix in another PR.

**Pull request merged**